nixos/modules/services/continuous-integration/github-runner.nix

```diff
@@ -118,6 +122,26 @@ in
       default = pkgs.github-runner;
       defaultText = literalExpression "pkgs.github-runner";
     };
+
+    ephemeral = mkOption {
+      type = types.bool;
+      description = ''
+        If enabled, causes the following behavior:
+
+        <itemizedlist>
+          <listitem><para>Passes the <literal>--ephemeral</literal> flag to the runner configuration script</para></listitem>
+          <listitem><para>De-registers and stops the runner with GitHub after it has processed one job</para></listitem>
+          <listitem><para>On stop, systemd wipes the runtime directory (this always happens, even without using the ephemeral option)</para></listitem>
+          <listitem><para>Restarts the service after its successful exit</para></listitem>
+          <listitem><para>On start, wipes the state directory and configures a new runner</para></listitem>
+        </itemizedlist>
+
+        You should only enable this option if <literal>tokenFile</literal> points to a file which contains a
+        personal access token (PAT). If you're using the option with a registration token, restarting the
+        service will fail as soon as the registration token expired.
+      '';
```

Comment:
Just curious: is there a reason you didn't use Markdown for this description?

Reply:
Done!